src/main/java/net/dv8tion/jda/api/entities/User.java

```diff
@@ -225,4 +228,168 @@ default String getEffectiveAvatarUrl()
+        
+        private final int offset;
+        private final long raw;
+        private final String name;
+
+        UserFlag(int offset, @Nonnull String name)
+        {
+            this.offset = offset;
+            this.raw = 1 << offset;
+            this.name = name;
+        }
+
+        /**
+         * The readable name as used in the Discord Client.
+         * 
+         * @return The readable name of this {@link net.dv8tion.jda.api.entities.User.UserFlag Flag}.
+         */
+        @Nonnull
+        public String getName()
+        {
+            return this.name;
+        }
+
+        /**
+         * The binary offset of the flag.
+         * 
+         * @return The offset that represents this {@link net.dv8tion.jda.api.entities.User.UserFlag Flag}.
+         */
+        public int getOffset()
+        {
+            return offset;
+        }
+
+        /**
+         * The value of this flag when viewed as raw value.
+         * <br>This is equivalent to: <code>1 {@literal <<} {@link #getOffset()}</code>
+         * 
+         * @return The raw value of this specific flag.
+         */
+        public long getRawValue()
+        {
+            return raw;
+        }
+
+        /**
+         * Gets the first {@link net.dv8tion.jda.api.entities.User.UserFlag Flag} relating to the provided offset.
+         * <br>If there is no {@link net.dv8tion.jda.api.entities.User.UserFlag Flag} that matches the provided offset,
+         * {@link net.dv8tion.jda.api.entities.User.UserFlag#UNKNOWN Flag.UNKNOWN} is returned.
+         * 
+         * @param  offset
+         *         The offset to match a {@link net.dv8tion.jda.api.entities.User.UserFlag Flag} to.
+         *         
+         * @return {@link net.dv8tion.jda.api.entities.User.UserFlag Flag} relating to the provided offset.
+         */
+        @Nonnull
+        public static UserFlag getFromOffset(int offset)
+        {
+            for (UserFlag flag : values())
+            {
+                if(flag.offset == offset)
+                    return flag;
+            }
+            return UNKNOWN;
+        }
+        
+        /**
+         * A set of all {@link net.dv8tion.jda.api.entities.User.UserFlag Flags} that are specified by this raw long representation of
+         * flags.
+         * 
+         * @param  flags
+         *         The raw {@code long} representation if flags.
+         *         
+         * @return Possibly-empty EnumSet of {@link UserFlag Flags}.
+         */
+        @Nonnull
+        public static EnumSet<UserFlag> getFlags(int flags)
+        {
+            if(flags == 0)
+                return EnumSet.noneOf(UserFlag.class);
+            EnumSet<UserFlag> userFlags = EnumSet.noneOf(UserFlag.class);
+            for (UserFlag flag : UserFlag.values())
+            {
+                if (flag != UNKNOWN && (flags & flag.raw) == flag.raw)
+                    userFlags.add(flag);
+            }
+            return userFlags;
```

Comment:
```suggestion
            final EnumSet<UserFlag> foundFlags = EnumSet.noneOf(UserFlag.class);
            
            if (flags == 0)
                return foundFlags; //empty
            
            for (UserFlag flag : values())
                if (flag != UNKNOWN && (flags & flag.raw) == flag.raw)
                    foundFlags.add(flag);
                    
            return foundFlags;
```

Not sure if this would help a bit, or if you want to keep the look of the method the exact same as in `Permissions`

Reply:
I also asked Minn why it is like that and not different (See [here](https://github.com/DV8FromTheWorld/JDA/pull/1271#issuecomment-619544061))
No idea why this comment was marked as off-topic.

I myself can't see a reason why `EnumSet.noneOf(Class)` is used twice as it could just be used once and modified later on...
I'll merge this suggestion and probably update the Permission class accordingly, unless Minn is against this.